### PR TITLE
Allow to insert association with predefined primary keys

### PR DESCRIPTION
**Pull request by nikolaistraessle**
If you have entities with predefined primary keys (e.g. UUID), then it should be possible to create this association on insert.

Check if an entity has a foreign key and if the foreign key value is zero, then insert a new row into the database. If the primary and the foreign key are set, update the given row.
If an entity has no foreign key, then check if the primary key is set or not.

Without this code change, it is not possible to insert the following struct:
```go
type A struct {
	Id string `json:"id" db:"id,primary"`
	B  B      `json:"b"`
}

type B struct {
	Id  string `json:"id" db:"id,primary"`
	AId string `json:"a_id" auto:"true"`
}
```

### CODE REVIEW

File: repository.go [hunk 1]
```diff
@@ -586,7 +586,8 @@ func (r repository) saveHasOne(cw contextWrapper, doc *Document, mutation *Mutat
 			assocMut         = assocMuts.Mutations[0]
 		)
 
-		if loaded {
+		if loaded && ("" == assoc.ForeignField() ||
```
Comment (Fs02) [inline repository.go:589]: please adjust the style: `assoc.ForeignField() == ""`

File: repository.go [hunk 2]
```diff
@@ -586,7 +586,8 @@ func (r repository) saveHasOne(cw contextWrapper, doc *Document, mutation *Mutat
...
+			("" != assoc.ForeignField() && !isZero(assoc.ForeignValue()))) {
```
Comment (Fs02) [inline repository.go:590]: isn't this enough? `loaded && (assoc.ForeignField() == "" || !isZero(assoc.ForeignValue()))`

Comment (nikolaistraessle) [reply repository.go:590]: @Fs02 you're right. I've fixed this. Many thanks for your review.

**Comment by Fs02** [comment]
Thanks for your contribution

**Pull request merged**